projects/table-builder/src/lib/directives/multi-sort.directive.ts

```diff
@@ -13,56 +14,40 @@ import { scan, filter, map } from 'rxjs/operators';
 })
 export class MultiSortDirective extends MatSort implements OnInit, OnDestroy {
   @Output() readonly multiSortChange: EventEmitter<Sort[]> = new EventEmitter<Sort[]>();
-
-  @Input() rules$: Observable<Sort[]> = new Subject();
-
-  @Input() set  reSort (val){
-    this.initRules();
-  } 
   rules: Sort[] = [];
-  preSortRules: Sort[] = [];
   private SubRef: Subscription;
 
+  constructor(private state: TableStateManager) {
+    super();
+  }
+
+  first = true;
+
   ngOnInit() {
-    this.SubRef = this.rules$.pipe(
-      scan((acc,curr)=>({prev: acc.curr, curr }),{prev: null, curr: null} as {prev: Sort[], curr: Sort[]}),
-      filter(acc => !acc.prev || acc.curr.some(sort => !acc.prev.some(prevSort => sort.active === prevSort.active && sort.direction === prevSort.direction))),
-      map(acc => acc.curr)
+    this.SubRef = this.state.state$.pipe(
+      map( state => state.sorted ),
+      distinctUntilChanged(),
     ).subscribe( rules => {
-      this.preSortRules = rules;
-      this.initRules();
+      this.rules = rules;
+      if(rules?.length > 0 && this.first) {
+        this.first = false;
+        this.active = this.rules[0].active;
+        this.direction = this.rules[0].direction;
+      }
+      this.multiSortChange.emit(this.rules);
```

Comment:
The whole `multiSortChange` can be removed since it does not get call on each change, no.

Reply:
removed